api2-war/src/main/java/eu/europeana/api2/v2/utils/ApiKeyUtils.java

```diff
@@ -22,9 +29,17 @@
 public class ApiKeyUtils {
 
     private static final Logger LOG = LogManager.getLogger(ApiKeyUtils.class);
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String UNABLETOVALIDATE = "Unable to validate apikey";
+    private static final String ERRORRETRIEVING = "Error retrieving apikey";
+    private static final String UNEXPECTED = "An unexpected error occurred during apikey validation";
+    private static final String TEMPKEY = "Temporary key";
 
     @Resource
-    private ApiKeyService apiService;
+    private ApiKeyService  apiService;
+
+    @Resource
+    private Api2UrlService urlService;
```

Comment:
I expected code in this class that checks whether we should query the the old or the new service. It seems right now both the ObjectController and QueryTranslationController will still use the old database. The idea was that we can switch to the new apikey service just by setting the apikey service url in the configuration.

Reply:
I was so involved in doing the changes in the ApikeyUtils etc. that I forgot that it's not just the searchcontroller which uses Apikey validation ... though the switch there was definitely being effected by adding the URL in the properties. The switching logic has now moved to ApikeyUtils, which makes it transparent from the pov of the (any) controller.